test/lib/code_corps/github/oauth_test.exs

```diff
@@ -0,0 +1,84 @@
+defmodule CodeCorps.GitHub.OAuthTest do
+  use CodeCorps.ModelCase
+
+  alias CodeCorps.{
+    GitHub, User
+  }
+
+  alias Ecto.Changeset
+
+  setup do
+    bypass = Bypass.open
+    GitHub.Bypass.setup(bypass)
+
+    on_exit fn ->
+      GitHub.Bypass.teardown()
+    end
+
+    {:ok, bypass: bypass}
+  end
+
+  describe "associate/2" do
+    test "updates the user, returns :ok tuple with user" do
+      user = insert(:user)
+      params = %{
+        github_auth_token: "foo_token",
+        github_avatar_url: "foo_url",
+        github_id: 123,
+        github_username: "foo_username"
+      }
+      {:ok, %User{} = returned_user} = GitHub.OAuth.associate(user, params)
+      assert user.id == returned_user.id
+      assert returned_user.github_auth_token == "foo_token"
+      assert returned_user.github_avatar_url == "foo_url"
+      assert returned_user.github_id == 123
+      assert returned_user.github_username == "foo_username"
+    end
+
+    test "returns :error tupple with changeset if there are validation errors" do
+      user = insert(:user)
+      params = %{}
+      {:error, %Changeset{} = changeset} = GitHub.OAuth.associate(user, params)
+      refute changeset.valid?
+    end
+  end
+
+  describe "connect/2" do
+    test "posts to github, updates user if reply is ok, returns updated user", %{bypass: bypass} do
+
+      Bypass.expect bypass, fn
+        %Plug.Conn{request_path: "/user"} = conn ->
+          response = %{
+            "avatar_url" => "foo_url",
+            "email" => "foo_email",
+            "id" => 123,
+            "login" => "foo_login"
+          }
+          Plug.Conn.resp(conn, 200, Poison.encode!(response))
+
+        conn ->
+          response = %{
+            "access_token" => "foo_auth_token"
+          }
+          Plug.Conn.resp(conn, 200, Poison.encode!(response))
+      end
```

Comment:
I am not super happy with how verbose I have to be here. I'd love for this to be reduced to a couple lines here somehow.